### PR TITLE
Make TDR & DSS plugin concurrency configurable; use 1 for TDR (#2281)

### PR DESCRIPTION
**Pull request by achave11**
Author (before primary review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `HCA` label <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes <sub>or this PR does not need porting</sub> 
- [x] Added `[r]` prefix at start of commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Ran `make requirements_update` <sub>or this PR does not change requirements*.txt</sub>
- [x] Mentioned `make requirements` in UPGRADING.rst <sub>or this PR does not change requirements*.txt</sub>
- [x] Documented upgrade of personal deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `[r]` prefix of commit title
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox` <sub>or commented that sandbox will be skipped</sub>
- [x] Reindexed `sandbox` <sub>or this PR does not require reindexing</sub>
- [x] Added PR reference to merge commit
- [x] Moved linked issue to Merged
- [x] Pushed merge commit to Github

Primary reviewer (after pushing merge commit)

- [x] Moved freebies to Merged column <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab
- [ ] Reindexed `dev` <sub>or this PR does not require reindexing</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Unassign reviewer from PR

### CODE REVIEW

File: environment.py
Comment (noah-aviel-dove) [inline environment.py:238]: Remove "/s". It makes no sense to describe the "number of worker", so there's no need for a separate grammar to cover  the case where the var is set to 1. It's also inconsistent with the documentation for the previous variable.

**Review by noah-aviel-dove — CHANGES_REQUESTED** [review]
Code changes look good, just a one tiny tweak to the documentation.

**Review by noah-aviel-dove — APPROVED** [review]

File: environment.py
Comment (hannes-ucsc) [inline environment.py:237]: TDR doesn't store metadata as files.

File: environment.py
Comment (hannes-ucsc) [inline environment.py:243]:
```suggestion
        # There is one such set of repository workers per index worker!
        # Using one worker as opposed to 8 (or 4)
        # improved the indexing time noticeably because it reduced
        # retries due to exceeding BigQuery's limit on the # of concurrent queries. Using two workers wasn't significantly faster.
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Comment by hannes-ucsc** [comment]
In English, the word after a semicolon is not capitalized.

**Comment by amarjandu** [comment]
@hannes-ucsc "I'd like to see Amar's changes on develop first so that's why this is blocking Abraham's PR."

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
Reminder to self: Don't merge before #2189.

File: environment.py
Comment (hannes-ucsc) [inline environment.py:238]:
```suggestion
        # There is one such set of repository workers per index worker.
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/6529

**Comment by hannes-ucsc** [comment]
I added a checklist item.

**Comment by achave11** [comment]
I ran the IT in my personal deployment, and it completed with no Errors or Failures.
When inspecting …
> https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/6529

… it seemed odd. I realize the sandbox deployment had not been updated since my branch was the last one pushed to GL.

**Comment by achave11** [comment]
I ran the IT in my personal deployment, and it completed with no Errors or Failures.

In regard to …
> https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/6529

… it seemed unusual, unlike something my changes would have caused. I realized that the `sandbox` deployment had not been updated since my branch was the last one to have been pushed to GL at the moment. I ran IT a second time and then it [succeeded](https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/6533).

**Comment by hannes-ucsc** [comment]
> it seemed unusual, unlike something my changes would have caused.

So what's the plan? Ignore the failure?

> I ran IT a second time and then it succeeded.

Next time, please ask before you touch `sandbox`.

**Comment by achave11** [comment]
> So what's the plan? Ignore the failure?

Not at all, I should have created a ticket. [This ticket](https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/6529) was created to investigate the failure.

> Next time, please ask before you touch sandbox.

Will do, my bad.

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
I've had to bounce back the blocker. Please hold this warm until the blocker lands.

**Comment by achave11** [comment]
Consolidated commits (removed 'fixup!' commit).

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
For demo, using CloudWatch logs compare the number of retries and the indexing time per bundle before and after this change.

**Pull request merged**
